tachyon/math/polynomials/combination/linear_combination.h

```diff
@@ -0,0 +1,142 @@
+#ifndef TACHYON_MATH_POLYNOMIALS_COMBINATION_LINEAR_COMBINATION_H_
+#define TACHYON_MATH_POLYNOMIALS_COMBINATION_LINEAR_COMBINATION_H_
+
+#include <algorithm>
+#include <functional>
+#include <memory>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+#include "absl/container/node_hash_map.h"
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/parallelize.h"
+#include "tachyon/crypto/sumcheck/sumcheck_proving_key.h"
+#include "tachyon/crypto/sumcheck/sumcheck_verifying_key.h"
+#include "tachyon/math/polynomials/combination/linear_combination_term.h"
+
+namespace tachyon {
+namespace math {
+
+template <typename MLE>
+class LinearCombination {
```

Comment:
When seeing the template typename it would be good to place it under `multivariate`! What do you think?